tests/integration/dynamic/python_types/__main__.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
+
+import binascii
+import os
+from pulumi import export, input_type, Input, output_type, Output
+from pulumi.dynamic import Resource, ResourceProvider, CreateResult
+
+@input_type
+class RandomSizeArgs:
+    size: Input[float]
+    def __init__(self, *, size: Input[float]):
+        self.size = size
+
+@output_type
+class RandomSize:
+    size: float
+
+class RandomResourceProvider(ResourceProvider):
+    def create(self, props):
+        size = int(props["size"]["size"])
```

Comment:
I don't quite follow this example - it seems to have an extra layer of nesting that isn't necessary.  Perhaps thats just because it's a particularly simple case - but it seems surprising that there needs to be an extra layer here.

Reply:
Yeah, this is a contrived example. I took the existing Python dynamic provider test and embellished it to use an input type, just to have some coverage that these would work with dynamic providers. I'll try to come up with a more realistic example of a dynamic provider that would actually use an input type in its design (that doesn't involve creating actual cloud resources).